### ISSUE TITLE
Adaptive gradient descent

### ISSUE THREAD

**Issue by terkkila** [terkkila/cgml#1]
Consider implementing Nesterov's accelerated gradient, momentum, and/or variance based adaptation of learning rates. This should make tuning of learning rate less critical to successful learning of model parameters.
